docs/user-guide/kubectl-overview.md

```diff
@@ -33,160 +33,261 @@ Documentation for other releases can be found at
 
 # kubectl overview
 
+Use this overview of the `kubectl` command line interface to help you start running commands against Kubernetes clusters. We will quickly cover `kubectl` syntax, describe the command operations, and provide common examples. For details about each command, including all the supported options, see the [kubectl](kubectl/kubectl.md) reference documentation.
+
 **Table of Contents**
 
-- [kubectl overview](#kubectl-overview)
-  - [Overview](#overview)
-  - [Common Operations](#common-operations)
-  - [Kubectl Operations](#kubectl-operations)
-  - [Resource Types](#resource-types)
+ * [Syntax](#syntax)
+ * [Operations](#operations)
+ * [Resource types](#resource-types)
+ * [Output options](#output-options)
+ * [Examples: Common operations](#examples-common-operations)
+
+TODO: Auto-generate this file to ensure it's always in sync with any `kubectl` changes, see [#14177](https://github.com/kubernetes/kubernetes/pull/14177).
+
+## Syntax
+
+Use the following syntax to run `kubectl` commands from your terminal window:
+
+```
+kubectl [command] [TYPE] [NAME] [options]
```

Comment:
@jlowdermilk Can you help me determine what files to update to get the edits here into the individual kubectl .md files?

(1)
Is this the file where the section titles are specified (specifically looking for "Options" and "Options inherited from parent commands" to change "Options" to "Flags")? [gen_kubectl_man.go](https://github.com/kubernetes/kubernetes/blob/master/cmd/genman/gen_kubectl_man.go)

(2)
To update the descriptions in each of the kubectl commands (so they match the edits in this file), do I update the individual kubectl-*`<operation>`*.1 files [here](https://github.com/kubernetes/kubernetes/tree/master/docs/man/man1)?